eng/scripts/patchhelpers.ps1

```diff
@@ -131,9 +133,6 @@ function CreateForwardLookingVersions($ArtifactInfos) {
 function FindAllArtifactsThatNeedPatching($ArtifactInfos, $AllDependenciesWithVersion) {
     foreach($arId in $ArtifactInfos.Keys) {
         $arInfo = $ArtifactInfos[$arId]
-        if($arInfo.GroupId -ne 'com.azure') {
-            continue;
-        }
 
```

Comment:
`ArtifactInfos` only include those from [`patch_release_client.txt`](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/patch_release_client.txt). I think it's safe to remove the check here.